### DISCUSSION TITLE
Error: No valid unarchiver found

### DISCUSSION THREAD

**Discussion by alekhyaaluru** [sparkle-project/Sparkle#2522] [Q&A]
### Description of the problem

I'm seeing a "No valid unarchiver found" error while trying to download and install an update using Sparkle

### Do you use Sandboxing in your app?
Yes
### Version of `Sparkle.framework` in the latest version of your app
2.5.2
### Version of `Sparkle.framework` in the old version of app that your users have (or N/A)
N/A
### Sparkle's output from Console.app
```
Error: An error occurred while extracting the archive. Please try again later. (null) (URL (null))
Error: No valid unarchiver was found for /Users/<username>/Library/Caches/<Test>/org.sparkle-project.Sparkle/Installation/ewZrYhRO2/login.html (null) (URL (null))
```

### Steps to reproduce the behavior

[The more information provided and pasted verbatim, the easier it will be to diagnose an issue. If you can provide the affected application/binary and XML feed to reproduce an issue, share them]

**Comment by zorgiepoo** [comment]
Likely an issue in your app. Notice how Sparkle downloaded `login.html` which is the wrong file. Maybe Sparkle does not have access to your download URL. If the download URL is not intended to be publicly accessible and you have some custom environment perhaps you'll need to set the httpHeaders on the SPUUpdater or something.

**Reply by alekhyaaluru** [reply]
Thanks for the response, I will try to check that

**Reply by alekhyaaluru** [reply]
is there a document/guide I can refer to, to set the httpHeaders? I am still new to Sparkle

**Reply by zorgiepoo** [reply]
https://sparkle-project.org/documentation/api-reference/Classes/SPUStandardUpdaterController.html#/c:objc(cs)SPUStandardUpdaterController(py)updater - getting the updater if using SPUStandardUpdaterController
https://sparkle-project.org/documentation/api-reference/Classes/SPUUpdater.html#/c:objc(cs)SPUUpdater(py)httpHeaders -setting httpHeaders

**Reply by alekhyaaluru** [reply]
I tried with a setup method as below to set the httpheaders but I'm still facing the same error. Can you please let me know if this is the right way to do it? 

`private func setupSparkle() {
        let token = "token token_here"
        let httpHeaders = ["Authorization": token]
        updaterController.updater.httpHeaders = httpHeaders
}`


Also is there a way to check what is actually getting downloaded to the target location?

**Reply by zorgiepoo** [reply]
Setting the httpHeaders looks right (not sure about the actual contents). I would set the httpHeaders right after starting the updater. (If you are instantiating the updater controller in a nib for a Cocoa app, then in the `applicationDidFinishLaunching:` should be good enough).

At the least you can see the filename being downloaded as you did in your first post here with the Console app.

I suppose (haven't tried this) if you set a breakpoint on the Obj-C selector `downloadDriverDidDownloadUpdate:` then the downloaded file should be somewhere in `/Library/Caches/<Test>/org.sparkle-project.Sparkle/`
